src/types.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use std::fmt;
+
+pub type LineAndColumnDisplay = text_lines::LineAndColumnDisplay;
+pub type LineAndColumnIndex = text_lines::LineAndColumnIndex;
+
+#[derive(Debug, Clone)]
+pub struct Diagnostic {
+  pub specifier: String,
+  pub display_position: LineAndColumnDisplay,
+  pub message: String,
+}
+
+impl std::error::Error for Diagnostic {}
+
+impl fmt::Display for Diagnostic {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    write!(
+      f,
+      "{} at {}:{}:{}",
+      self.message,
+      self.specifier,
+      self.display_position.line_number,
+      self.display_position.column_number
+    )
+  }
+}
+
+// Warning! The values in this enum are duplicated from the TypeScript compiler and
```

Comment:
I don't think this warning holds anymore, though I am the one who put it there and we keep copying it around.  Now we transform into the TypeScript internal enum versus sending the media type value into the compiler.

Reply:
As stated in the `deno_graph` PR, I think we should just move `MediaType` into here wholly and make it part of `deno_ast` and re-export it from `deno_graph`.

Reply:
Both done now and agree.